python/pyspark/testing/connectutils.py

```diff
@@ -58,7 +60,18 @@ class PlanOnlyTestFixture(unittest.TestCase):
 
     connect: "MockRemoteSession"
 
+    @classmethod
+    def _read_table(cls, table_name: str) -> "DataFrame":
+        return DataFrame.withPlan(Read(table_name), cls.connect)
+
+    @classmethod
+    def _udf_mock(cls, *args, **kwargs):
+        return "internal_name"
+
     @classmethod
     def setUpClass(cls: Any) -> None:
         cls.connect = MockRemoteSession()
         cls.tbl_name = f"tbl{uuid.uuid4()}".replace("-", "")
+
+        cls.connect.set_hook("register_udf", cls._udf_mock)
+        cls.connect.set_hook("readTable", cls._read_table)
```

Comment:
I think we should also drop the table and udf at `teardownClass`.

Reply:
makes sense. done